packages/aws-cdk-lib/core/lib/cfn-resource.ts

```diff
@@ -644,66 +609,11 @@ function deepMerge(target: any, ...sources: any[]) {
 
     for (const key of Object.keys(source)) {
       const value = source[key];
-      if (typeof(value) === 'object' && value != null && !Array.isArray(value)) {
+      if (typeof(value) === 'object' && value != null && !Array.isArray(value) && !(value instanceof Intrinsic)) {
```

Comment:
`instanceof` is not safe in all contexts. See this [PR](https://github.com/aws/aws-cdk/pull/20564/files) for how we do this in other places.

Reply:
I don't understand why instanceof should not be used here. The PR has no description and it is not even merged. Why do you think so?